WooCommerce/Classes/Blaze/BlazeLocalNotificationScheduler.swift

```diff
@@ -0,0 +1,166 @@
+                                                         ascending: false)
+        let resultsController = ResultsController<StorageBlazeCampaignListItem>(storageManager: storageManager,
+                                                                                matching: predicate,
+                                                                                sortedBy: [sortDescriptorByStartTime])
+        return resultsController
+    }()
+
+    init(siteID: Int64,
+         stores: StoresManager = ServiceLocator.stores,
+         storageManager: StorageManagerType = ServiceLocator.storageManager,
+         userDefaults: UserDefaults = .standard,
+         pushNotesManager: PushNotesManager = ServiceLocator.pushNotesManager,
+         blazeEligibilityChecker: BlazeEligibilityCheckerProtocol = BlazeEligibilityChecker()) {
+        self.siteID = siteID
+        self.stores = stores
+        self.storageManager = storageManager
+        self.pushNotesManager = pushNotesManager
+        self.scheduler = LocalNotificationScheduler(pushNotesManager: pushNotesManager)
+        self.userDefaults = userDefaults
+        self.blazeEligibilityChecker = blazeEligibilityChecker
+    }
+
+    /// Starts observing campaigns from storage and schedules local notification
+    ///
+    func scheduleNotifications() async {
+        guard await isEligibleForBlaze() else {
+            DDLogDebug("Blaze: Store not eligible for Blaze. Don't schedule local notification.")
+            return
+        }
+
+        observeStorageAndScheduleNotifications()
+
+        pushNotesManager.localNotificationUserResponses
+            .sink { [weak self] response in
+                guard let self,
+                      response.notification.request.identifier == LocalNotification.Scenario.blazeNoCampaignReminder.identifier else {
+                    return
+                }
+
+                userDefaults.setBlazeNoCampaignReminderOpened(true)
+            }
+            .store(in: &subscriptions)
+    }
+}
+
+private extension DefaultBlazeLocalNotificationScheduler {
+    func isEligibleForBlaze() async -> Bool {
+        guard let site = stores.sessionManager.defaultSite else {
+            return false
+        }
+        return await blazeEligibilityChecker.isSiteEligible(site)
+    }
+
+    /// Performs initial fetch from storage and updates results.
+    func observeStorageAndScheduleNotifications() {
+        blazeCampaignResultsController.onDidChangeContent = { [weak self] in
+            self?.scheduleLocalNotificationIfNeeded()
+        }
+        blazeCampaignResultsController.onDidResetContent = { [weak self] in
+            self?.scheduleLocalNotificationIfNeeded()
+        }
+
+        do {
+            try blazeCampaignResultsController.performFetch()
+            scheduleLocalNotificationIfNeeded()
+        } catch {
+            ServiceLocator.crashLogging.logError(error)
```

Comment:
If an error is generated, are we going to know that it's specifically related to this section?

Minor nit:
```suggestion
                    ServiceLocator.crashLogging.logError(error, message: "Failed to perform fetch for Blaze campaigns")
```

Reply:
Thanks for the suggestion, Paolo! But, the `logError` method doesn't accept a message parameter. 🤔

Reply:
👋 Paolo, I am enabling auto-merge to get this into the next release. I will address any further comments in future PR.